scripts/dockerlib.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Author Gabriel Hartmann <gabhart@microsoft.com>
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Open Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+
+file_name=${0##*/}
+
+timestamp() {
+    date +"%Y-%m-%d %H:%M:%S"
```

Comment:
I'll remove the timestamp function.  The principle that trivially short, only called once, functions shouldn't exist sounds good to me.  Most of the functions below don't fall into that category as they're called multiple times.  Some I just moved to a common place because I thought it made sense, e.g. the *_dump set of functions should be together.  Others I kept for the sake of neatness.  For example get_script_dir is called 5 times, get_config_dir 2 times, but get_log_dir only 1 time.  They all do roughly the same thing so I thought it made sense to have all of them.  Even those can be refactored further since they have mostly copy pasted content, but refactoring can go too far.

Let me know if you want to undo some of these changes.  In any case I'll get rid of the timestamp function.